File: backend/src/main/java/me/performancereservation/domain/user/entitiy/ManagerRequest.java
package me.performancereservation.domain.user.entitiy;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import me.performancereservation.domain.common.BaseEntity;
import me.performancereservation.domain.user.enums.ManagerRequestStatus;

import java.time.LocalDate;


@Entity
@Getter
@NoArgsConstructor
public class ManagerRequest extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id; // 매니저 권한 승인 요청 내역 ID

    private Long userId; // (FK) 유저 ID

    private String organizationName;        // 개인 혹은 단체명
    private String organizationContact;     // 연락처
    private String experience;              // 공연 경험
    private String reason;                  // 신청 사유

    @Enumerated(EnumType.STRING)
    private ManagerRequestStatus status; // 승인 요청 상태

    private LocalDate approvedAt; // 승인 날짜

    @Builder
    public ManagerRequest(Long id,
                          Long userId,
                          String reason,
                          String experience,
                          LocalDate approvedAt,
                          String organizationName,
                          String organizationContact,
                          ManagerRequestStatus status) {
        this.id = id;
        this.userId = userId;
        this.status = status;
        this.reason = reason;
        this.approvedAt = approvedAt;
        this.experience = experience;
        this.organizationName = organizationName;
        this.organizationContact = organizationContact;
    }

    public boolean isPending() {
        return status == ManagerRequestStatus.PENDING;
    }

    public void approve() {
        status = ManagerRequestStatus.APPROVED;
        approvedAt = LocalDate.now();
    }

    public void reject() {
        status = ManagerRequestStatus.REJECTED;
    }
}

File: backend/src/main/java/me/performancereservation/domain/user/entitiy/ManagerRequest.kt
package me.performancereservation.domain.user.entitiy

import jakarta.persistence.*

import me.performancereservation.domain.common.BaseEntity
import me.performancereservation.domain.user.enums.ManagerRequestStatus
import java.time.LocalDate

@Entity
class ManagerRequest(// 매니저 권한 승인 요청 내역 ID
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    val userId: Long,

    val reason: String,                  // 신청 사유

    val experience: String,              // 공연 경험

    val organizationName: String,        // 개인 혹은 단체명

    val organizationContact: String,     // 연락처
    @Column
    var approvedAt: LocalDate? = null,    // 승인 날짜

    @Enumerated(EnumType.STRING)
    var status: ManagerRequestStatus    // 승인 요청 상태
) : BaseEntity() {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is ManagerRequest) return false
        if (id == null || other.id == null) return false
        return id == other.id && updatedAt == other.updatedAt
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + updatedAt.hashCode()
        return result
    }

    val isPending: Boolean
        get() = status == ManagerRequestStatus.PENDING

    fun approve() {
        status = ManagerRequestStatus.APPROVED
        approvedAt = LocalDate.now()
    }

    fun reject() {
        status = ManagerRequestStatus.REJECTED
    }
}
